AuthenticationAndAuthorization/AuthenticationAndAuthorization/Endpoints/GenreEndpoints.cs

```diff
@@ -0,0 +1,24 @@
+﻿using AuthenticationAndAuthorization.Policy;
+using BusinessLogic.Interfaces;
+using BusinessLogic.Model.GenreModel;
+using DataAccounts.Entitys.GenreEntitys;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AuthenticationAndAuthorization.Endpoints
+{
+    public static class GenreEndpoints
+    {
+        public static void MapGenreEndpoints(this IEndpointRouteBuilder endpoints)
+        {
+            endpoints.MapPost("genres", ([FromServices] IGenreService genreService, [FromBody] CreateGenreModel genre)
+                => genreService.CreateGenre(genre)).RequireAuthorization(Policies.AdminPolicy);
+
+            endpoints.MapDelete("genres/{id:int}", ([FromServices] IGenreService genreService, [FromBody] DeleteGenreModel genre)
+                => genreService.DeleteGenre(genre)).RequireAuthorization(Policies.AdminPolicy);
+
+            endpoints.MapPost("genres/{filter}", ([FromServices] IGenreService genreService, [FromBody] GetGenresListModel filterModel)
```

Comment:
```suggestion
            endpoints.MapGet("genres", ([FromServices] IGenreService genreService, [FromQuery] GetGenresListModel filterModel)
```

to fetch data we use GET verb and pass filters as query params